src/plugins/tooltip/index.ts

```diff
@@ -71,22 +75,54 @@ class HSTooltip extends HSBasePlugin<{}> implements ITooltip {
 	}
 
 	private click() {
-		if (this.el.classList.contains('show')) return false;
-
-		this.show();
-
-		const handle = () => {
-			setTimeout(() => {
-				this.hide();
+		// Close the current tooltip if one is open
+		if (HSTooltip.currentTooltip && HSTooltip.currentTooltip !== this) {
+			HSTooltip.currentTooltip.hide();
+		}
 
-				this.toggle.removeEventListener('click', handle, true);
-				this.toggle.removeEventListener('blur', handle, true);
+		// Toggle the visibility of the current tooltip
+		if (this.el.classList.contains('show')) {
+			// If already open, hide it
+			this.hide();
+			HSTooltip.currentTooltip = null; // Clear the current tooltip reference
+		} else {
+			// Otherwise, show it
+			this.show();
+
+			const handle = () => {
+				setTimeout(() => {
+					this.hide();
+					document.body.removeEventListener('click', handle); // Remove the event listener
+					document.body.removeEventListener('blur', handle);  // Remove the event listener
+				});
+			};
+
+			document.body.addEventListener('click', handle);
+			document.body.addEventListener('blur', handle); 
+
+			// Add an event listener to the tooltip content to prevent closing when clicked inside
+			// With this when click inside content it will not close
+			// If you want to close it when click inside content you can remove this part
+			// The link or button inside content will work as expected
+			this.content?.addEventListener('click', (event) => {
+				event.stopPropagation(); // Prevent the click event from bubbling up
 			});
-		};
+			
 
-		this.toggle.addEventListener('click', handle, true);
-		this.toggle.addEventListener('blur', handle, true);
+			// Update the current tooltip reference
+			HSTooltip.currentTooltip = this;
+		}
 	}
+	
+	/**
+	 * Handles outside click events to hide the popover if clicked outside.
+	 * Used to ensure the popover is hidden when clicking outside of it.
+	 */
+	private handleOutsideClick = (event: MouseEvent) => {
+        if (!this.el.contains(event.target as Node)) {
```

Comment:
Hello @olegpix ,

Thank you for your feedback. I appreciate it. I haven't come across any errors during my usage.

Upon reviewing the video below, it appears that the trigger is activated by a click event. We can integrate a similar click event within the popover, utilizing either a button or a link, to ensure seamless functionality.


https://github.com/htmlstreamofficial/preline/assets/110020770/35027785-d20a-4376-bc5c-ab844d33a099





Could you please specify when this error occurred so that I can address it?